drivers/serial/uart_ns16550.c

```diff
@@ -99,6 +100,21 @@ BUILD_ASSERT(IS_ENABLED(CONFIG_PCIE), "NS16550(s) in DT need CONFIG_PCIE");
 #define PCP_UPDATE 0x80000000 /* update clock */
 #define PCP_EN 0x00000001     /* enable clock output */
 
+/* Fields for TI K3 UART module */
+
+#define MDR1_MODE_SELECT_FIELD_MASK		BIT_MASK(3)
+#define MDR1_MODE_SELECT_FIELD_SHIFT		BIT_MASK(0)
+
+/* Modes available for TI K3 UART module */
+
+#define MDR1_STD_MODE					(0)
+#define MDR1_UART_16X					(1)
+#define MDR1_UART_13X					(2)
+#define MDR1_MIR_MODE					(3)
+#define MDR1_FIR_MODE					(4)
+#define MDR1_CIR_MODE					(5)
+#define MDR1_DISABLE					(6)
```

Comment:
Why are these modes so different to Linux?

	#define UART_OMAP_MDR1_16X_MODE		0x00	/* UART 16x mode */
	#define UART_OMAP_MDR1_SIR_MODE		0x01	/* SIR mode */
	#define UART_OMAP_MDR1_16X_ABAUD_MODE	0x02	/* UART 16x auto-baud */
	#define UART_OMAP_MDR1_13X_MODE		0x03	/* UART 13x mode */
	#define UART_OMAP_MDR1_MIR_MODE		0x04	/* MIR mode */
	#define UART_OMAP_MDR1_FIR_MODE		0x05	/* FIR mode */
	#define UART_OMAP_MDR1_CIR_MODE		0x06	/* CIR mode */
	#define UART_OMAP_MDR1_DISABLE		0x07	/* Disable (default state) */

https://github.com/torvalds/linux/blob/master/include/uapi/linux/serial_reg.h#L368

Reply:
I have taken the modes from the TRM for the AM62x, page 8961.

https://www.[ti.com/lit/ug/spruiv7a/spruiv7a.pdf](https://www.ti.com/lit/ug/spruiv7a/spruiv7a.pdf)

Reply:
Those seem to align better with the Linux modes:

	7 | MODE_SELECT_Value_7 | Disable (default state)
	6 | MODE_SELECT_Value_6 | CIR mode
	5 | MODE_SELECT_Value_5 | FIR mode
	4 | MODE_SELECT_Value_4 | MIR mode
	3 | MODE_SELECT_Value_3 | UART 13x mode
	2 | MODE_SELECT_Value_2 | UART 16x auto-baud
	1 | MODE_SELECT_Value_1 | SIR mode

Strange that value `0` is not included in the TRM

Reply:
Oh okay, I think I might have misread the values. Will update with this, and leave 0 as standard. Thank you for the suggestion.